### PR TITLE
feat: add backward compatibility for autoware_auto_msgs and autoware_msgs

### PR DESCRIPTION
**Pull request by kminoda**
## Description
Support both `autoware_auto_msgs` and `autoware_msgs`

## How to review
See the pytest results in CI.

## Reference
None

## Notes for reviewer
None

### CODE REVIEW

**Comment by kminoda** [comment]
@ktro2828 Hi, would you review this again?

File: perception_dataset/rosbag2/autoware_msgs.py [hunk 1]
```diff
@@ -129,18 +140,29 @@ def get_category_name(classification: List[ObjectClassification]) -> str:
...
+        if isinstance(obj, AutowareAutoDetectedObject) or isinstance(obj, AutowareDetectedObject):
             obj_uuid = uuid.uuid4()  # random uuid
             velocity: Dict[str, float] = {
                 "x": obj.kinematics.twist_with_covariance.twist.linear.x,
                 "y": obj.kinematics.twist_with_covariance.twist.linear.y,
                 "z": obj.kinematics.twist_with_covariance.twist.linear.z,
             }
             acceleration: Optional[Dict[str, float]] = None
-        elif isinstance(obj, TrackedObject):
+        elif isinstance(obj, AutowareAutoTrackedObject) or isinstance(obj, AutowareTrackedObject):
```
Comment (ktro2828) [inline perception_dataset/rosbag2/autoware_msgs.py:165]: we can make this shorten
```suggestion
        elif isinstance(obj, (AutowareAutoTrackedObject, AutowareTrackedObject)):
```

File: perception_dataset/rosbag2/autoware_msgs.py [hunk 2]
```diff
@@ -129,18 +140,29 @@ def get_category_name(classification: List[ObjectClassification]) -> str:
...
+        ]
         pose = obj.kinematics.pose_with_covariance.pose
 
-        if isinstance(obj, DetectedObject):
+        print(
+            AutowareAutoDetectedObject,
+            AutowareAutoTrackedObject,
+            AutowareDetectedObject,
+            AutowareTrackedObject,
+        )
```
Comment (ktro2828) [inline perception_dataset/rosbag2/autoware_msgs.py:156]: [Q] What is the purpose of this `print`? Can we remove it?

```suggestion
```

File: perception_dataset/rosbag2/autoware_msgs.py [hunk 3]
```diff
@@ -129,18 +140,29 @@ def get_category_name(classification: List[ObjectClassification]) -> str:
...
+        if isinstance(obj, AutowareAutoDetectedObject) or isinstance(obj, AutowareDetectedObject):
```
Comment (ktro2828) [inline perception_dataset/rosbag2/autoware_msgs.py:157]: we can make this shorten.
```suggestion
        if isinstance(obj, (AutowareAutoDetectedObject, AutowareDetectedObject)):
```

File: perception_dataset/rosbag2/autoware_msgs.py [hunk 4]
```diff
@@ -2,12 +2,17 @@
...
 )
+from autoware_auto_perception_msgs.msg import DetectedObject as AutowareAutoDetectedObject
+from autoware_auto_perception_msgs.msg import DetectedObjects as AutowareAutoDetectedObjects
+from autoware_auto_perception_msgs.msg import TrackedObject as AutowareAutoTrackedObject
+from autoware_auto_perception_msgs.msg import TrackedObjects as AutowareAutoTrackedObjects
+from autoware_perception_msgs.msg import DetectedObject as AutowareDetectedObject
+from autoware_perception_msgs.msg import DetectedObjects as AutowareDetectedObjects
+from autoware_perception_msgs.msg import ObjectClassification as AutowareObjectClassification
+from autoware_perception_msgs.msg import TrackedObject as AutowareTrackedObject
+from autoware_perception_msgs.msg import TrackedObjects as AutowareTrackedObjects
```
Comment (ktro2828) [inline perception_dataset/rosbag2/autoware_msgs.py:15]: According to updated `README`, users can install one of `autoware_auto_perception_msgs` or `autoware_perception_msgs`. Then will `ImportError` be caused if users do it?

**Review by ktro2828 — COMMENTED** [review]
Thanks for your PR :rocket: ! I left some comments.

File: perception_dataset/rosbag2/autoware_msgs.py (hunk 1 above)
Comment (kminoda) [reply perception_dataset/rosbag2/autoware_msgs.py]: Thanks. Fixed: https://github.com/tier4/tier4_perception_dataset/pull/110/commits/41ada9161c023e4cb2e22865820e7377708cb80b

File: perception_dataset/rosbag2/autoware_msgs.py (hunk 3 above)
Comment (kminoda) [reply perception_dataset/rosbag2/autoware_msgs.py]: Thanks. Fixed: https://github.com/tier4/tier4_perception_dataset/pull/110/commits/41ada9161c023e4cb2e22865820e7377708cb80b

File: perception_dataset/rosbag2/autoware_msgs.py (hunk 2 above)
Comment (kminoda) [reply perception_dataset/rosbag2/autoware_msgs.py]: Seems I forgot to remove debugging print. Fixed: https://github.com/tier4/tier4_perception_dataset/pull/110/commits/9ffbd3aa4be03a3619a9bd17acd74b67b22ce9a3

File: perception_dataset/rosbag2/autoware_msgs.py (hunk 4 above)
Comment (kminoda) [reply perception_dataset/rosbag2/autoware_msgs.py:15]: Sorry the readme was misleading. Fixed: https://github.com/tier4/tier4_perception_dataset/pull/110/commits/436c6ae4b57f8d576b84b6bc09dde7da86301f87

(The basic idea is to install both libraries from build_depend.repos, but it either of the package already exists in the user's environment, the may remove either of them to build only the necessary one)

**Review by ktro2828 — APPROVED** [review]
Code looks good to me and confirmed that all tests were passed.

**Pull request merged**